File: core/java/src/main/java/org/aiddl/core/java/function/misc/ValueFunction.java
package org.aiddl.core.java.function.misc;

import org.aiddl.core.java.interfaces.Function;
import org.aiddl.core.java.representation.Term;

public class ValueFunction implements Function {

	@Override
	public Term apply(Term x) {
		return x.getValue();
	}
}

File: common/scala/src/main/scala/org/aiddl/common/scala/optimization/ValueFunction.scala
package org.aiddl.common.scala.optimization

import org.aiddl.core.scala.representation.*

trait ValueFunction[S, N] extends Ordering[S] {
  val ordering: Ordering[N]
  def apply(x: S): N

  override def compare(x: S, y: S): Int =
    ordering.compare(this(x), this(y))
}
